### PR TITLE
fix(core): values in edit not saved when closing info modal

### PR DESCRIPTION
**Pull request by JimmFly**
Before the modification, when the modal is closed, if a value is being edited, the input's on blur event cannot be triggered to automatically save.

https://github.com/toeverything/AFFiNE/assets/102217452/60235cec-0022-4c4d-b213-28f2331a0c5b

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#7474** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7474?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7465** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7465?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx [hunk 1]
```diff
@@ -63,14 +75,19 @@ export const InfoModal = ({
...
-      onOpenChange={onOpenChange}
+      onOpenChange={handleOpenChange}
       withoutCloseButton
     >
       <Scrollable.Root>
         <Scrollable.Viewport
           className={styles.viewport}
           data-testid="info-modal"
         >
+          <input
```
Comment (pengx17) [inline packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx]: sorry I don't understand this fix. Can you please explain?

Comment (JimmFly) [reply packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx]: Before the modification, when the modal is closed, if a value is being edited, the input's on blur event cannot be triggered to automatically save.
The way I handle this issue is to create a hidden input and automatically focus it before closing the modal, thus triggering on blur event for the other input.

File: packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx [hunk 2]
```diff
@@ -39,7 +39,19 @@ export const InfoModal = ({
   workspace: Workspace;
 }) => {
   const titleInputHandleRef = useRef<InlineEditHandle>(null);
+  const hiddenInputRef = useRef<HTMLInputElement>(null);
+
   const manager = usePagePropertiesManager(page);
+
+  const handleOpenChange = useCallback(
+    (open: boolean) => {
+      if (!open) {
```
Comment (CatsJuice) [inline packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx]: Maybe just execute `(document.activeElement as HTMLElement)?.blur()` here?

File: packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx [hunk 3]
```diff
@@ -63,14 +75,19 @@ export const InfoModal = ({
...
     >
       <Scrollable.Root>
         <Scrollable.Viewport
           className={styles.viewport}
           data-testid="info-modal"
         >
+          <input
+            type="text"
+            ref={hiddenInputRef}
+            className={styles.hiddenInput}
```
Comment (CatsJuice) [inline packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx]: [`hidden`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden)

File: packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx (hunk 1 above)
Comment (pengx17) [reply packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx]: I see. Not sure if this fix is too hacky. Maybe adding an exiting animation could also work?

Comment (CatsJuice) [reply packages/frontend/core/src/components/affine/page-properties/info-modal/info-modal.tsx]: If animation is necessary, I think it's better to implement in global `<Modal />` and make it configurable.

**Review by EYHN — APPROVED** [review]

**Pull request merged**